src/pages/programDisaggregations/form/CategoryMapping.tsx

```diff
@@ -0,0 +1,187 @@
+}
+
+type MappingNameModalProps = {
+    onSave: (mappingName: string) => void
+    closeModal: () => void
+    originalName?: string
+}
+
+const MappingNameModal = ({
+    onSave,
+    closeModal,
+    originalName,
+}: MappingNameModalProps) => {
+    const [mappingName, setMappingName] = useState<string>(originalName ?? '')
+    return (
+        <Modal position="middle" onClose={closeModal}>
+            <ModalTitle>{i18n.t('Rename category mapping')}</ModalTitle>
+            <ModalContent>
+                <Input
+                    initialFocus={true}
+                    value={mappingName}
+                    onChange={(e) => {
+                        setMappingName(e.value ?? '')
+                    }}
+                ></Input>
+            </ModalContent>
+            <ModalActions>
+                <ButtonStrip>
+                    <Button onClick={closeModal}>{i18n.t('Cancel')}</Button>
+                    <Button
+                        onClick={() => {
+                            onSave(mappingName)
+                        }}
+                    >
+                        {i18n.t('Apply')}
+                    </Button>
+                </ButtonStrip>
+            </ModalActions>
+        </Modal>
+    )
+}
+const CategoryMappingWrapper = ({
+    deleted,
+    children,
+}: {
+    deleted?: boolean
+    children: ReactNode
+}) => {
+    return (
+        <div className={deleted ? css.mappingCardDeleted : css.mappingCard}>
+            {children}
+        </div>
+    )
+}
+
+type CategoryMappingProps = {
+    fieldName: string
+    categoryOptionArray: CategoryOption[]
+    showSoftDelete: boolean
+}
+export const CategoryMapping = ({
+    fieldName,
+    categoryOptionArray,
+    showSoftDelete = true,
+}: CategoryMappingProps) => {
+    const categoryMapping = useField(fieldName)
+    const categoryMappingOnChange = categoryMapping?.input?.onChange
+    const categoryMappingValue = categoryMapping?.input?.value
+    const categoryOptionInformation = categoryOptionArray.reduce(
+        (optionsToName: Record<string, string>, opt) => {
+            return { ...optionsToName, [opt.id]: opt.displayName }
+        },
+        {} as Record<string, string>
+    )
+    const [nameModalOpen, setNameModalOpen] = useState(false)
+    const closeModal = useCallback(() => {
+        setNameModalOpen(false)
+    }, [setNameModalOpen])
+    const saveName = useCallback(
+        (newName: string) => {
+            categoryMappingOnChange({
+                ...categoryMappingValue,
+                mappingName: newName,
+            })
+            closeModal()
+        },
+        [categoryMappingOnChange, categoryMappingValue, closeModal]
+    )
+    if (categoryMappingValue.deleted) {
+        return (
+            <CategoryMappingWrapper deleted>
+                <div className={css.deletedMappingText}>
+                    {i18n.t(
+                        '{{- categoryMappingName}} mapping will be deleted on save',
```

Comment:
This I think is okay? i18n generally encodes special characters to avoid injection attacks. When we trust the string (it's from the backend), it's an easy choice to override the encoding. In this case, the user is updating the name, but I think the general protection in jsx against malicious strings should be enough?